src/kixi/eventlog/application.clj

```diff
@@ -19,36 +21,43 @@
 
 (defn topic-definitions [max-message-size topic-names]
   (zipmap topic-names
-          (map (fn [topic-name] 
+          (map (fn [topic-name]
                  {:num-partitions     (Integer/valueOf
                                        (or (System/getenv (str (.toUpperCase topic-name) "_TOPIC_NUM_PARTITIONS"))
-                                           "3"))
-                  :replication-factor (Integer/valueOf 
+                                           "1"))
+                  :replication-factor (Integer/valueOf
                                        (or (System/getenv (str (.toUpperCase topic-name) "_TOPIC_REPLICATION_FACTOR"))
-                                           "2") )
+                                           "1") )
                   :max-message-size   max-message-size}) topic-names)))
 
 (defn parse-topics [topics]
   (re-seq #"[\w-]+" topics))
 
+(defn config
+  [profile]
+  (try (aero/read-config (io/resource "eventlog.edn") {:resolver aero/relative-resolver :profile profile})
+       (catch java.io.FileNotFoundException _ (log/info "no authentication config found"))))
+
 (defn new-system
-  ([] (let [zookeeper-connect (or (System/getenv "ZK_CONNECT") "localhost:2181")
-            max-message-size  (or (System/getenv "TOPIC_MAX_MESSAGE_SIZE") "1000000")
-            producer          (new-producer :max-message-size max-message-size)
-            topic-names       (or (System/getenv "TOPICS")
-                                  "flight_events01 hotel_events01")
-            topics            (new-topics 
-                               (topic-definitions max-message-size 
-                                                  (parse-topics topic-names)))]
-        (-> (map->EventLogApi
-            {:web-server  (web/new-server)
-             :repl-server (Object.) ; dummy - replaced when invoked via controller.main
-             :zookeeper   (new-zk-client zookeeper-connect)
-             :topics      topics
-             :producer    producer})
-           (component/system-using
-            {:producer   [:zookeeper]
-             :topics     [:zookeeper]
-             :web-server [:producer :topics]}))))
-  ([extra-components]
-     (merge (new-system) extra-components)))
+  ([{:keys [profile authentication]}]
+   (let [zookeeper-connect (or (System/getenv "ZK_CONNECT") "localhost:2181")
+         max-message-size  (or (System/getenv "TOPIC_MAX_MESSAGE_SIZE") "1000000")
+         producer          (new-producer :max-message-size max-message-size)
+         topic-names       (or (System/getenv "TOPICS")
+                               "flight_events01 hotel_events01")
```

Comment:
I don't think we need these as defaults in the code any more or we can choose other default values.